### PR TITLE
Swift 2.2 support

### PR DESCRIPTION
**Pull request by ikesyo**
~~This is another breaking change because of the `associatedtype` usage.~~

Updated `Nimble` to be built with Xcode 7.3.

### CODE REVIEW

**Comment by mdiep** [comment]
Would it be possible to maintain compatibility with both Swift 2.1 and Swift 2.2 for a single release? :pray:

**Comment by ikesyo** [comment]
> Would it be possible to maintain compatibility with both Swift 2.1 and Swift 2.2 for a single release? :pray:

Okay, let's keep using `typealias` atm, that is deprecated but still can be used.

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,24 @@
 language: objective-c
-osx_image: xcode7.2
 script: placeholder # workaround for https://github.com/travis-ci/travis-ci/issues/4681
 matrix:
   exclude:
     - script: placeholder # workaround for https://github.com/travis-ci/travis-ci/issues/4681
   include:
     - script: script/cibuild
+      osx_image: xcode7.2
```
Comment (NachoSoto) [inline .travis.yml]: Nice

**Comment by mdiep** [comment]
:joy: 

This is great stuff. :clap: Thanks for doing the work! :sparkling_heart:

**Pull request merged**